airbyte-integrations/connectors/source-intercom/source_intercom/components.py

```diff
@@ -175,186 +139,52 @@ def __post_init__(self, parameters: Mapping[str, Any]):
+        if self.parent_stream_name in stream_state and stream_state.get(self.parent_stream_name, {}).get(self.parent_cursor_field):
+            parent_stream_state = {self.parent_cursor_field: stream_state[self.parent_stream_name][self.parent_cursor_field]}
+            self._state[self.parent_stream_name] = parent_stream_state
+            if "prior_state" in self._state:
+                self._state["prior_state"][self.parent_stream_name] = parent_stream_state
+
+    def close_slice(self, stream_slice: StreamSlice, most_recent_record: Optional[Record]) -> None:
+        super().close_slice(stream_slice=stream_slice, most_recent_record=most_recent_record)
+        if self.parent_stream:
+            self._state[self.parent_stream_name] = self.parent_stream.state
+
+    def stream_slices(self) -> Iterable[Mapping[str, Any]]:
+        parent_state = (self._state or {}).get(self.parent_stream_name, {})
+        slices_generator = self.read_parent_stream(self.parent_sync_mode, self.parent_cursor_field, parent_state)
+        yield from [slice for slice in slices_generator] if self.parent_complete_fetch else slices_generator
 
     def read_parent_stream(
         self, sync_mode: SyncMode, cursor_field: Optional[str], stream_state: Mapping[str, Any]
     ) -> Iterable[Mapping[str, Any]]:
+        self.parent_stream.state = stream_state
 
-        for parent_slice in self.parent_stream.stream_slices(sync_mode=sync_mode, cursor_field=cursor_field, stream_state=stream_state):
-            empty_parent_slice = True
+        parent_stream_slices_gen = self.parent_stream.stream_slices(
+            sync_mode=sync_mode, cursor_field=cursor_field, stream_state=stream_state
+        )
 
-            # update slice with parent state, to pass the initial parent state to the parent instance
-            # stream_state is being replaced by empty object, since the parent stream is not directly initiated
-            parent_prior_state = self._cursor.get("prior_state", {}).get(self.parent_stream_name, {}).get(self.parent_cursor_field)
-            parent_slice.update({"prior_state": {self.parent_cursor_field: parent_prior_state}})
+        for parent_slice in parent_stream_slices_gen:
 
-            for parent_record in self.parent_stream.read_records(
+            parent_records_gen = self.parent_stream.read_records(
                 sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=parent_slice, stream_state=stream_state
-            ):
-                # Skip non-records (eg AirbyteLogMessage)
-                if isinstance(parent_record, AirbyteMessage):
-                    if parent_record.type == Type.RECORD:
-                        parent_record = parent_record.record.data
+            )
 
+            for parent_record in parent_records_gen:
                 try:
-                    substream_slice = dpath.util.get(parent_record, self.parent_field)
+                    substream_slice_value = parent_record[self.parent_field]
```

Comment:
can we use `get` method logic to avoid` try...except `?

Reply:
For me the use of `try...except` block here is a little bit more clear. But in case of performance it will take longer to handle an exception block if a parent record does not contain a key. Performance should allways be on first place so I agree to switch `try...except` to `if` statement